packages/block-editor/src/components/block-list/use-block-props/use-block-refs.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useReducer } from '@wordpress/element';
+
+const store = {
```

Comment:
This is a global store that's not going to work well in a multi  block editor instances. a Provider would be better here IMO.

Reply:
Hm, I'm not following. Client IDs are unique in the whole application?

Reply:
they are indeed but I personally think we should avoid global style in singletons like that as much as we can, this is the exact use case for context providers. 

I think this kind of thing create issues for npm consumption (deduplication of modules...)

Reply:
Ok, I'll look into it

Reply:
@youknowriad How would you solve the "skip to selected block" component, which would be rendered outside of the context tree? Somehow it needs access to the right store. Should we use `useSelect` for these refs and store them in the some store as the blocks?

Reply:
> How would you solve the "skip to selected block" component, which would be rendered outside of the context tree?

What is this component? 

> Somehow it needs access to the right store. Should we use useSelect for these refs and store them in the some store as the blocks?

This is a solution but if we do this, this hook is useless, we can just keep the block node in the store and use useEffect if we need to react to changes

Reply:
This one: https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/skip-to-selected-block/index.js